op-node/rollup/derive/engine_controller.go

```diff
@@ -307,6 +309,6 @@ func (e *EngineController) ForkchoiceUpdate(ctx context.Context, state *eth.Fork
 }
 
 // NewPayload implements LocalEngineControl.
-func (e *EngineController) NewPayload(ctx context.Context, payload *eth.ExecutionPayload) (*eth.PayloadStatusV1, error) {
-	return e.engine.NewPayload(ctx, payload)
+func (e *EngineController) NewPayload(ctx context.Context, payload *eth.ExecutionPayload, parentBeaconBlockRoot *common.Hash) (*eth.PayloadStatusV1, error) {
```

Comment:
Why doesn't new payload take the envelope?

Reply:
Because the Engine API RPC definition of NewPayload does it this way